librz/asm/p/asm_tms320c64x.c

```diff
@@ -54,12 +54,11 @@ static int tms320c64x_disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int len
 		goto fin;
 	}
 	op->size = insn->size;
-	char *buf_asm = rz_str_newf("%s%s%s", insn->mnemonic, insn->op_str[0] ? " " : "", insn->op_str);
-	if (buf_asm) {
-		rz_str_replace_char(buf_asm, '%', 0);
-		rz_str_case(buf_asm, false);
-		rz_asm_op_set_asm(op, buf_asm);
-		free(buf_asm);
+	rz_asm_op_setf_asm(op, "%s%s%s", insn->mnemonic, insn->op_str[0] ? " " : "", insn->op_str);
+	char *str = rz_asm_op_get_asm(op);
+	if (str) {
+		rz_str_replace_char(str, '%', 0);
```

Comment:
@wargio I think it isn't a good idea to modify content of RzStrBuf in that way.

Reply:
We should avoid unnecessary generalization in this path though - it's a "hot" one, since it runs for every instruction.

Reply:
1) RzStrBuf doesn't allocate buffer for short strings
2) Length field is not modified